src/util/result.h

```diff
@@ -39,13 +39,25 @@ class Result
 
     std::variant<bilingual_str, T> m_variant;
 
+    //! Disallow copy constructor, require Result to be moved for efficiency.
+    Result(const Result&) = delete;
+
+    //! Disallow operator= to avoid confusion in the future when the Result
+    //! class gains support for richer error reporting, and callers should have
+    //! ability to set a new result value without clearing existing error
+    //! messages.
+    Result& operator=(const Result&) = default;
+    Result& operator=(Result&&) = default;
```

Comment:
Can you briefly explain why these `operator=` are made private as opposed to deleting them?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/29906#discussion_r1581409638

> Can you briefly explain why these `operator=` are made private as opposed to deleting them?

Changed these to `delete`. The default operators were needed in an earlier version of the PR which had an `Update()` method which called them privately.